### PR TITLE
CLN: Simplify rolling.py helper functions

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Cleans helper functions in `rolling.py` and remove dead code paths.

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -305,17 +290,6 @@ def _wrap_result(self, result, block=None, obj=None):
 
         if isinstance(result, np.ndarray):
 
-            # coerce if necessary
-            if block is not None:
-                if is_timedelta64_dtype(block.values.dtype):
```
Comment (jbrockmendel) [inline pandas/core/window/rolling.py]: is this not reachable?

**Comment by jbrockmendel** [comment]
Nice cleanup

File: pandas/core/window/rolling.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/window/rolling.py]: Don't think so. Our test suite includes timedelta64 which doesn't his this branch.

File: pandas/core/window/common.py [hunk 2]
```diff
@@ -250,31 +250,10 @@ def _get_center_of_mass(comass, span, halflife, alpha):
     return float(comass)
 
 
-def _offset(window, center):
+def _calculate_center_offset(window):
```
Comment (jreback) [inline pandas/core/window/common.py]: as a followup typing & doc-strings would be great. also ok to de-privatize these

File: pandas/core/window/rolling.py [hunk 3]
```diff
@@ -1051,12 +1019,12 @@ def _get_window(
             # GH #15662. `False` makes symmetric window, rather than periodic.
             return sig.get_window(win_type, window, False).astype(float)
 
-    def _get_weighted_roll_func(
-        self, cfunc: Callable, check_minp: Callable, **kwargs
-    ) -> Callable:
-        def func(arg, window, min_periods=None, closed=None):
-            minp = check_minp(min_periods, len(window))
-            return cfunc(arg, window, minp, **kwargs)
+    @staticmethod
```
Comment (jreback) [inline pandas/core/window/rolling.py]: instead of static can you make this a module level function, or is there a reason you want this as static?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/common.py [hunk 4]
```diff
@@ -336,3 +315,12 @@ def _prep_binary(arg1, arg2):
     Y = arg2 + 0 * arg1
 
     return X, Y
+
+
+def get_weighted_roll_func(cfunc: Callable) -> Callable:
```
Comment (WillAyd) [inline pandas/core/window/common.py]: Subtypes for Callable would be helpful here if you know from looking at this what they are

**Review by WillAyd — APPROVED** [review]
Lgtm

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke 

doc-strings and types always welcome

**Pull request merged**